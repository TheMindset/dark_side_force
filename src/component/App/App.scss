@import url('https://fonts.googleapis.com/css?family=Merriweather&display=swap');

* {
  box-sizing: border-box;
  font-family: 'Merriweather', monospace;
  margin: 0;
  padding: 0;
}

.App {
  background-attachment: fixed;
  background: url('../Images/background.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  height: 120vh;
}