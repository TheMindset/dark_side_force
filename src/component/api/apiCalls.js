export const getFilms = () => {
  return fetch('https://swapi.co/api/films')
  .then(response => {
    if (!response.ok) {
      throw new Error(`Error: ${response.status}, please retry`)
    }
    return response.json()
  })
  .then(data => data.results)
  .then(allFilms => sortFilmsByEpisode(allFilms))
  .catch(error => console.log(error))
}

const sortFilmsByEpisode = (allFilms) => {
  return allFilms.sort((a, b) => a.episode_id - b.episode_id)
  .map(film => {
    return {
      title: film.title,
      episode: film.episode_id,
      releaseYear: film.release_date.slice(0, 4)
    }
  })
} 