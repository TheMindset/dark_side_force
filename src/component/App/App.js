import React, { Component } from 'react';
import { getFilms } from '../api/apiCalls'
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      movies: []
    }
  }

  componentDidMount() {
    getFilms()
    .then(data => this.setState({ movies: data }))
  }

  render() {
    return (
      <div>
        <h1>The Dark Side Force</h1>
      </div>
    )
  }
}
export default App;
