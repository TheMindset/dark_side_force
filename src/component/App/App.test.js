import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
import { getFilms, getCharacters } from '../../Util/api/apiCalls'

jest.mock('../../Util/api/apiCalls')

describe('App', () => {
  let wrapper, mockFilms

  beforeEach(() => {
    mockFilms = [
      {
        title: 'The Avengers',
        episode: 1,
        releaseYear: 2012,
        id: 1
      },
      {
        title: 'Avengers: Age of Ultron',
        episode: 2,
        releaseYear: 2015,
        id: 2
      }  
    ]

    getFilms.mockImplementation(() => {
      return Promise.resolve(mockFilms)
    })
    wrapper = shallow(<App />)
  })

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should retreive films after mounting', () => {
    shallow(<App />)

    expect(getFilms).toHaveBeenCalled()
  })

  test('should update the user state when getFormData is called', () => {
    const mockUserInfos = {
      name: 'Tony Startk',
      quote: 'The king in north',
      rank: 'Dead'
    }

    window.localStorage = jest.fn()

    expect(wrapper.state('userInfos')).toEqual({})
    expect(wrapper.state('isFormComplete')).toEqual(false)
    wrapper.instance().getFormData(mockUserInfos)
    expect(wrapper.state('userInfos')).toEqual(mockUserInfos)
    expect(wrapper.state('isFormComplete')).toEqual(true)
  })  
})
