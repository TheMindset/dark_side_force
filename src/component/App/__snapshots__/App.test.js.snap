// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="App"
>
  <div>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
  </div>
</main>
`;

exports[`App with nav should render Nav if the form is completed 1`] = `
<main
  className="App"
>
  <div>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
  </div>
</main>
`;

exports[`App with no characters should render Scroll if no characters is present 1`] = `
<main
  className="App"
>
  <div>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
  </div>
</main>
`;

exports[`App with no movies should render loading images if no movies present  1`] = `
<main
  className="App"
>
  <div>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
  </div>
</main>
`;
